### ISSUE TITLE
Neither contacts nor calendars are migrated from the 8.2 compatible apps which caues complete data loss

### ISSUE THREAD

**Issue by BernhardPosselt** [owncloud/core#22883]
### Steps to reproduce
1. Upgrade a fully working 8.2 with calendar and contacts apps to 9.0
2. Use the new calendar and contacts apps since the old ones are not compatible with 9.0
3. All your calendars and contacts are gone

What happens is that the schema from the new calendar and contacts apps do not contain the caldav/carddav tables anymore, since those are in core now. As a result, those tables are dropped and all your contacts and calendar entries are lost. Prior to upgrade there is no migration of any sort.

Even worse: you can't even import vcards in the new contacts apps which makes it impossible to migrate anything without complete data loss (apart from manual sql INSERTs)

TL;DR: people using contacts and calendar will lose all data when upgrading to 9.0 and contacts can not even be migrated manually

Blocker?

@georgehrke @jancborchardt @karlitschek @jospoortvliet @DeepDiver1975 @PVince81 @LukasReschke @Henni

**Comment by karlitschek** [comment]
which new contacts and calendar di you mean? you updated to the new ones direectly frim the github repo for 8.2? they should be marked as compatible with 9.0 or newer only. 
@deepdiver1975 @georgehrke

**Comment by tomneedham** [comment]
I believe this is what the two dav:migrate commands are for: https://github.com/owncloud/core/blob/master/apps/dav/command/migratecalendars.php and https://github.com/owncloud/core/blob/master/apps/dav/command/migrateaddressbooks.php

**Comment by BernhardPosselt** [comment]
@tomneedham tables are dropped before you can migrate since both apps dont ship the old database.xml -> database schema migration drops the tables
@karlitschek all I did was a git pull from owncloud/contacts and owncloud/calendar

**Comment by tomneedham** [comment]
@BernhardPosselt Yup I saw your comment in IRC. I think @DeepDiver1975 knows the full story here.

**Comment by BernhardPosselt** [comment]
Shame on me, migrate command worked :)

**Comment by karlitschek** [comment]
hmm. you shouldn't do that but install the ones from the app store which handles the right dependencies. not sure how you managed to upgrade to the new calendar in owncloud 8.2 because the dependencies in info.xml say 9 or newer

**Comment by RealRancor** [comment]
> I believe this is what the two dav:migrate commands are for: https://github.com/owncloud/core/blob/master/apps/dav/command/migratecalendars.php and https://github.com/owncloud/core/blob/master/apps/dav/command/migrateaddressbooks.php

Seems thats currently not documented that this is needed. Have raised https://github.com/owncloud/documentation/issues/2215 for that. cc @carlaschroder

**Comment by RealRancor** [comment]
PR for a release note in https://github.com/owncloud/documentation/issues/2216. Feedback welcome

**Comment by DeepDiver1975** [comment]
> tables are dropped before you can migrate since both apps dont ship the old database.xml -> database schema migration drops the tables

No tables are removed - we shall not destroy data! :wink:

**Comment by DeepDiver1975** [comment]
> Shame on me, migrate command worked :)

Good to have the commands but this should be done automatically on upgrade - has happened before https://github.com/owncloud/core/issues/22678

Let me reopen this to have a closer look

**Comment by RealRancor** [comment]
@DeepDiver1975 So https://github.com/owncloud/documentation/pull/2216 is not needed?

**Comment by DeepDiver1975** [comment]
> @DeepDiver1975 So owncloud/documentation#2216 is not needed?

It's okay to document this - but we better state that the migration is done automatically and the occ command is a fallback for failed updates or an option to test migration

**Comment by RealRancor** [comment]
Thanks! Have updated https://github.com/owncloud/documentation/pull/2216 accordingly.

**Comment by MorrisJobke** [comment]
@DeepDiver1975 See https://github.com/owncloud/documentation/pull/2216#issuecomment-192964842 - `// the occ tool is *for now* only available in debug mode for developers to test`

**Comment by PVince81** [comment]
When I upgraded my server from 8.2.2 to 9.0 RC1 I had the old calendar + contacts app enabled.
After the upgrade, the calendars were successfully moved to the new tables and the apps got disabled.

Also, @nickvergessen has reported that the automatic upgrade also worked for him.

**Comment by BernhardPosselt** [comment]
BTW, when migrating calendars manually (dav:migrate-calendars myadminuser) this failed for me until ive executed dav:migrate-calendars nonadminuser

**Comment by BernhardPosselt** [comment]
Not sure if its related to admins though and my test data has been migrated

**Comment by nickvergessen** [comment]
> As a result, those tables are dropped and all your contacts and calendar entries are lost.

As stated in IRC, this is wrong. No tables are dropped.

> Prior to upgrade there is no migration of any sort.

There is and it worked for vinc and me.
Can you confirm, so we can close this?

**Comment by nickvergessen** [comment]
Also your bug report does not contain the necessary data (apps list, etc) to continue any investigation

**Comment by BernhardPosselt** [comment]
@nickvergessen feel free to close it however @DeepDiver1975 reopened it, please contact him before

**Comment by enoch85** [comment]
@DeepDiver1975 Will this be done automatically, and will all the users have to change their addresses on their clients i.e. DAVDroid to ../remote.php/dav instead of the old one?

**Comment by DeepDiver1975** [comment]
> and will all the users have to change their addresses on their clients (i.e. DAVDroid) to ../remote.php/dav instead of the old one?

the old as well as the new dav url will work

**Comment by DeepDiver1975** [comment]
I guess we can close this

**Comment by enoch85** [comment]
Thanks! But still, will this be done automatically? @DeepDiver1975 

@BernhardPosselt Did you have to do it manually?

**Comment by DeepDiver1975** [comment]
> But still, will this be done automatically?

yes

**Comment by Andreacroci** [comment]
For what it's worth: I upgraded to 9.0 yesterday using the steps in https://doc.owncloud.org/server/9.0/admin_manual/maintenance/package_upgrade.html. I did have some problems, but at the end it worked. However, the calendar did not get upgraded automatically. I don't know exactly what happened to the app itself, because I then downloaded the app from the stores where it says it's compatible with 9.0 and reinstalled it before checking what happened to the other one.

At any rate, all my calendar events are lost: the just don't show up. Amazingly enough the contacts are still there. So apparently it's not that automatic yet.

A quick manual SELECT on oc_calendarobjects shows the data are still in the db (mysql), but for some reason they are not shown.

No idea how to pull them back up.

**Comment by RealRancor** [comment]
> If only calendar is missing make sure that you're not using IE/Edge: owncloud/calendar#309

**Comment by Andreacroci** [comment]
All right. No, I'm using Chrome under Ubuntu 14.04.3 LTS, where also OC 
is installed using Apache 2.4, PHP 5.6 and mySQL 5.7.

Any other information you need, please just let me know.

**Comment by Andreacroci** [comment]
Sorry, I forgot mentioning something that may be useful:

I installed the calendar app by downloading the version for OC 9.0 from 
here: https://apps.owncloud.com/content/show.php/Calendar?content=168707 
and putting it in the apps/ folder. I double checked the permissions, 
that are ok.

When I run the check-code on it, I get the following:

Duplicate min ownCloud version requirement found
Deprecated field available: require => 9
App is not compliant

**Comment by RealRancor** [comment]
@Andreacroci Thats unrelated to this issue and should be reported to the issue tracker of the calendar app.

**Comment by georgehrke** [comment]
> Deprecated field available: require => 9

Was already fixed, please don't report

**Comment by Andreacroci** [comment]
OK, I won't report, but where do I find the app with the fixed code then?

**Comment by georgehrke** [comment]
in the calendar repo, this error should not stop you from using your version though

**Comment by georgehrke** [comment]
If there is data in oc_calendars and oc_calendarobjects and you can access it with caldav clients via http://owncloud.dev/remote.php/dav/, but you can't see them in the calendar app, report an issue in the calendar repo please.

**Comment by Andreacroci** [comment]
No, I do have data in oc_calendars and oc_calendarobjects, but I can't 
access it via caldav client either: I have tried with both Thunderbird 
and DAVDroid from two smartphones. Thunderbird gives me the little 
yellow exclamation mark saying "The calendar owncloud is momentarily not 
available". DAVDroid tells me there is no carddav or caldav service at 
that address.

Now I checked again and the contacts are gone too. Not from the 
database, but from the WebGUI. That's weird: they were there before and 
the database is accessible by the owncloud user. I guess I need to do 
some more troubleshooting.
